### PR TITLE
Feature/ego integration

### PR DESCRIPTION
**Pull request by lepsalex**
EGO/SONG Support - Tag new JWT based strategy enhancement in progress server #239
EGO/SONG Support - Create JWT Strategy enhancement in progress server #238 
EGO/SONG Support - Unit Tests for Existing and New Security in progress server test #237

### CODE REVIEW

**Comment by lepsalex** [comment]
ok to test

**Comment by lepsalex** [comment]
add to whitelist

File: song-server/src/test/java/org/icgc/dcc/song/server/security/StudyScopeStrategyTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package org.icgc.dcc.song.server.security;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
```
Comment (andricDu) [inline song-server/src/test/java/org/icgc/dcc/song/server/security/StudyScopeStrategyTest.java]: unused

File: song-server/src/test/java/org/icgc/dcc/song/server/security/StudyScopeStrategyTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.springframework.boot.SpringBootConfiguration;
```
Comment (andricDu) [inline song-server/src/test/java/org/icgc/dcc/song/server/security/StudyScopeStrategyTest.java]: Unused

File: song-server/src/test/java/org/icgc/dcc/song/server/security/StudyScopeStrategyTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import org.springframework.security.authentication.TestingAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.OAuth2Request;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.TestPropertySource;
```
Comment (andricDu) [inline song-server/src/test/java/org/icgc/dcc/song/server/security/StudyScopeStrategyTest.java]: unused

**Comment by andricDu** [comment]
add to whitelist

**Comment by andricDu** [comment]
test this please

**Comment by andricDu** [comment]
test this please

**Comment by andricDu** [comment]
test this please

**Comment by andricDu** [comment]
test this please

**Comment by andricDu** [comment]
SONG is now using API Tokens issued by Ego rather than the JWTs that were issued to Ego. 

Closing this PR as this feature is no longer required.

**Pull request closed without merging**